qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -273,13 +273,14 @@ def _compute_weighted_kernel_sum(self, index: int, X: np.ndarray, training: bool
             else:
                 kernel_value = X[index, j]
 
-            value += (
+            value += cast(
+                float,
                 # alpha weights the contribution of the associated datum
                 self._alphas[j]
                 # the class membership labels have to be in {-1, +1}
                 * self._label_map[self._y_train[j]]
                 # the offset to the kernel function leads to an implicit bias term
-                * (kernel_value + self._kernel_offset)
+                * (kernel_value + self._kernel_offset),
             )
```

Comment:
Not quite sure why `cast` is required. Tried locally with the latest `mypy` installed, don't see problems without this cast.

Reply:
it requires with those new mypy settings

Reply:
I've checked out the PR, updated `mypy`, looks good without `cast`.

Reply:
I will try in the CI without it then because it fails locally for me.

Reply:
You are right. I didn't have `sparse` installed. Once I installed it, it didn't require the cast. I removed it in this PR. Thanks for double checking it.

Reply:
Oh, that's very strange, there's nothing that really depends on `sparse`.

Reply:
Looking further when I installed `sparse`, it downgraded my numpy to 1.20.3 from 1.21.5. Reinstalling 1.21.5 caused mypy to require the cast  again. Going back to 1.20.3, it works.

Reply:
OK, I see. So, perhaps this problem will arise again when we upgrade `numpy`.